### DISCUSSION TITLE
How do I make a property read-only in a DTO?

### DISCUSSION THREAD

**Discussion by Peter-Svahn** [RicoSuter/NSwag#4300] [Q&A]
I'm using Nswag to generate a client API for a UserController, where UserDTO objects are defined as:

public class UserDTO
{
    public int Id { get; set; }
    public DateTime CreatedTime { get; private set; }
    public DateTime UpdatedTime { get; private set; }
    public string? UserName { get; set; }
    public string? DisplayName { get; set; }
    public string? EncodedPassword { get; set; }
    public string? Email { get; set; }
    public bool Active { get; set; }
    public List<Role>? Roles { get; set; }
}

I would like to disallow clients to set created and update time by making the setters private. But the generated UserDTO on the client side still has a public setter. Are there any attributes to control the OpenAPI document generation in this respect?

**Comment by Franck-Costner** [comment]
Hello,

I have the same issue, could someone provide an update ?
I have already try with **ReadOnly=true** and **private set;**

Thanks in advance
